control/strategy/pool.go

```diff
@@ -270,8 +270,9 @@ func (p *pool) Eligible() bool {
 		return false
 	}
 
-	should := p.SubscriptionCount() / p.concurrencyCount
-	if should > p.Count() && should <= p.max {
+	should := p.SubscriptionCount() > p.concurrencyCount*p.Count()
+	// Check if pool is eligible and number of plugins is less than maximum allowed
+	if should && p.Count() < p.max {
```

Comment:
I'm wondering if we should change the code at 269 to be:
```go
if p.Count() >= p.max {
        return false
}
``` 
This would capture the case where plugin count went over the max and ensure that when we get to the next statement p.Count() < p.max. Then we could modify this if-statement to check only:
```go
if p.SubscriptionCount() > p.ConcurrencyCount*p.Count() {
       return true
}
```